lib/commands/timeout.js

```diff
@@ -13,6 +14,20 @@ commands.timeouts = async function (name, duration) {
   }
 };
 
+commands.implicitWaitForCondition = async function (condFn) {
+  log.debug("Waiting up to " + this.implicitWaitMs + "ms for condition");
+  let wrappedCondFn = async function (...args) {
+    // reset command timeout
+    // TODO: can we do better than that?
+    if (this.noCommandTimer) {
+      clearTimeout(this.noCommandTimer);
+    }
+    return await condFn(...args);
+  }.bind(this);
+  return await waitForCondition(wrappedCondFn, {
```

Comment:
It's better, less chance of regenerator bugs, all the return promises are the same.